core/src/main/java/io/hyperfoil/core/handlers/json/JsonParser.java

```diff
@@ -376,8 +376,24 @@ public void parse(ByteStream data, Session session, boolean isLast) {
                stream.consume(lastOutputIndex, safeOutputIndex, record, this, session, isLast);
                lastOutputIndex = safeOutputIndex;
             }
-            stream.release(readerIndex);
+            stream.releaseUntil(readerIndex);
+         }
+      }
+
+      private int safeReleaseIndex() {
```

Comment:
While fixing this, I've tried to improve `StreamQueue::releaseUntil` to be able to remove the part if the `index` is exactly the last addressable one in the part, but it has broke few tests :"( so I'm not sure yet if change that behaviour and picked a more conservative direction :/ - beaten by the code complexity